core/test_network/src/lib.rs

```diff
@@ -598,9 +601,9 @@ impl PeerBuilder {
     pub fn build(&mut self) -> Result<Peer> {
         let mut peer = Peer::new()?;
         if let Some(port) = self.port.take() {
-            peer.p2p_address = format!("127.0.0.1:{port}");
-            peer.api_address = format!("127.0.0.1:{}", port + 1);
-            peer.telemetry_address = format!("127.0.0.1:{}", port + 2);
+            peer.p2p_address = socket_addr!(127,0,0,1; port);
```

Comment:
Any particular reason why the syntax is not `127.0.0.1:port`? Should we have a ticket to rewrite as proc-macro that uses the standard syntax?

Reply:
Because 127.0 is a float, so standard syntax can't be expressed as a `macro_rules` macro.
I didn't think this was important enough to warrant a proc-macro, but I could rewrite it as one if you think it's a good idea

Reply:
I'd relegate it to a separate issue.